lte/gateway/deploy/roles/magma/files/set_irq_affinity

```diff
@@ -0,0 +1,265 @@
+fi
+
+if [ -z "$XPS_ENA" ]; then
+	XPS_ENA=$XPS_DIS
+fi
+
+num='^[0-9]+$'
+# Vars
+AFF=$1
+shift
+
+case "$AFF" in
+    remote)	[[ $1 =~ $num ]] && rnode=$1 && shift ;;
+    one)	[[ $1 =~ $num ]] && cnt=$1 && shift ;;
+    all)	;;
+    local)	;;
+    custom)	;;
+    [0-9]*)	;;
+    -h|--help)	usage ;;
+    "")		usage ;;
+    *)		IFACES=$AFF && AFF=all ;;	# Backwards compat mode
+esac
+
+# append the interfaces listed to the string with spaces
+while [ "$#" -ne "0" ] ; do
+	IFACES+=" $1"
+	shift
+done
+
+# for now the user must specify interfaces
+if [ -z "$IFACES" ]; then
+	usage
+	exit 1
+fi
+
+# support functions
+
+set_affinity()
+{
+	VEC=$core
+	if [ $VEC -ge 32 ]
+	then
+		MASK_FILL=""
+		MASK_ZERO="00000000"
+		let "IDX = $VEC / 32"
+		for ((i=1; i<=$IDX;i++))
+		do
+			MASK_FILL="${MASK_FILL},${MASK_ZERO}"
+		done
+
+		let "VEC -= 32 * $IDX"
+		MASK_TMP=$((1<<$VEC))
+		MASK=$(printf "%X%s" $MASK_TMP $MASK_FILL)
+	else
+		MASK_TMP=$((1<<$VEC))
+		MASK=$(printf "%X" $MASK_TMP)
+	fi
+
+	printf "%s" $MASK > /proc/irq/$IRQ/smp_affinity
+	printf "%s %d %s -> /proc/irq/$IRQ/smp_affinity\n" $IFACE $core $MASK
+	case "$XPS_ENA" in
+	1)
+		printf "%s %d %s -> /sys/class/net/%s/queues/tx-%d/xps_cpus\n" $IFACE $core $MASK $IFACE $((n-1))
+		printf "%s" $MASK > /sys/class/net/$IFACE/queues/tx-$((n-1))/xps_cpus
+	;;
+	2)
+		MASK=0
+		printf "%s %d %s -> /sys/class/net/%s/queues/tx-%d/xps_cpus\n" $IFACE $core $MASK $IFACE $((n-1))
+		printf "%s" $MASK > /sys/class/net/$IFACE/queues/tx-$((n-1))/xps_cpus
+	;;
+	*)
+	esac
+}
+
+# Allow usage of , or -
+#
+parse_range () {
+        RANGE=${@//,/ }
+        RANGE=${RANGE//-/..}
+        LIST=""
+        for r in $RANGE; do
+		# eval lets us use vars in {#..#} range
+                [[ $r =~ '..' ]] && r="$(eval echo {$r})"
+		LIST+=" $r"
+        done
+	echo $LIST
+}
+
+# Affinitize interrupts
+#
+setaff()
+{
+	CORES=$(parse_range $CORES)
+	ncores=$(echo $CORES | wc -w)
+	n=1
+
+	# this script only supports interrupt vectors in pairs,
+	# modification would be required to support a single Tx or Rx queue
+	# per interrupt vector
+
+	queues="${IFACE}-.*TxRx"
+
+	irqs=$(grep "$queues" /proc/interrupts | cut -f1 -d:)
+	[ -z "$irqs" ] && irqs=$(grep $IFACE /proc/interrupts | cut -f1 -d:)
+	[ -z "$irqs" ] && irqs=$(for i in `ls -Ux /sys/class/net/$IFACE/device/msi_irqs` ;\
+	                         do grep "$i:.*TxRx" /proc/interrupts | grep -v fdir | cut -f 1 -d : ;\
+	                         done)
+	[ -z "$irqs" ] && echo "Error: Could not find interrupts for $IFACE"
+
+	echo "IFACE CORE MASK -> FILE"
+	echo "======================="
+	for IRQ in $irqs; do
+		[ "$n" -gt "$ncores" ] && n=1
+		j=1
+		# much faster than calling cut for each
+		for i in $CORES; do
+			[ $((j++)) -ge $n ] && break
+		done
+		core=$i
+		set_affinity
+		((n++))
+	done
+}
+
+# now the actual useful bits of code
+
+# these next 2 lines would allow script to auto-determine interfaces
+#[ -z "$IFACES" ] && IFACES=$(ls /sys/class/net)
```

Comment:
rm ?

Reply:
I basically copied this script from gist, so I would like to keep it as it is for easier diff in future.